Source/Scene/CameraFlightPath.js

```diff
@@ -410,136 +262,93 @@ define([
         }
-        if ((defined(direction) && !defined(up)) || (defined(up) && !defined(direction))) {
-            throw new DeveloperError('If either direction or up is given, then both are required.');
-        }
         //>>includeEnd('debug');
 
+        var projection = scene.mapProjection;
+        var ellipsoid = projection.ellipsoid;
+
+        var maximumHeight = options.maximumHeight;
+        var easingFunction = options.easingFunction;
+
         if (scene.mode === SceneMode.MORPHING) {
-            return {
-                startObject : {},
-                stopObject: {},
-                duration : 0.0
-            };
+            return emptyFlight();
         }
 
         var convert = defaultValue(options.convert, true);
 
         if (convert && scene.mode !== SceneMode.SCENE3D) {
-            var projection = scene.mapProjection;
-            var ellipsoid = projection.ellipsoid;
             ellipsoid.cartesianToCartographic(destination, scratchCartographic);
             destination = projection.project(scratchCartographic, scratchDestination);
         }
 
-        var duration = defaultValue(options.duration, 3.0);
-        var controller = scene.screenSpaceCameraController;
-        controller.enableInputs = false;
-
-        var wrapCallback = function(cb) {
-            var wrapped = function() {
-                if (typeof cb === 'function') {
-                    cb();
-                }
-
-                controller.enableInputs = true;
-            };
-            return wrapped;
-        };
-        var complete = wrapCallback(options.complete);
-        var cancel = wrapCallback(options.cancel);
-
         var camera = scene.camera;
         var transform = options.endTransform;
         if (defined(transform)) {
             camera._setTransform(transform);
         }
 
-        var frustum = camera.frustum;
-        if (scene.mode === SceneMode.SCENE2D) {
-            if (Cartesian2.equalsEpsilon(camera.position, destination, CesiumMath.EPSILON6) && (CesiumMath.equalsEpsilon(Math.max(frustum.right - frustum.left, frustum.top - frustum.bottom), destination.z, CesiumMath.EPSILON6))) {
-                return {
-                    startObject : {},
-                    stopObject: {},
-                    duration : 0.0,
-                    complete : complete,
-                    cancel: cancel
-                };
-            }
-        } else if (Cartesian3.equalsEpsilon(destination, camera.position, CesiumMath.EPSILON6)) {
-            return {
-                startObject : {},
-                stopObject: {},
-                duration : 0.0,
-                complete : complete,
-                cancel: cancel
-            };
+        var duration = options.duration;
+        if (!defined(duration)) {
+            duration = Math.ceil(Cartesian3.distance(camera.position, destination) / 1000000.0) + 3.0;
+            duration = Math.min(duration, 10.0);
```

Comment:
This line seems to be the cause of the extremely slow fights that I'm seeing (10 seconds).  I would suggest changing these two lines to the below, which I feel works much better in all of the test cases I've tried.

```
            duration = Math.ceil(Cartesian3.distance(camera.position, destination) / 1000000.0) + 1.0;
            duration = Math.min(duration, 2.0);
```